### PR TITLE
Added lots of features

### PR DESCRIPTION
**Pull request by augustak**
Changelog

- Added createAppClient (POST /appClients)
- Added listAppClients (GET /appClients)
- Added deleteAppClient (DELETE /appClients/:id)
- Added deleteAsset (DELETE /assets/:id)
- Added listBatches (GET /batches)
- Added deleteBatch (DELETE /batches/:id)
- Updated deleteDocuments (DELETE /documents) to use options including batchId, consentId, maxResults and nextToken 
- Added listLogs (GET /logs)
- Added deleteSecret (DELETE /secrets/:id)
- Updated CreateTransitionOptions to use dedicated types for DockerTransitionParameters and ManualTransitionParameters
- Added completedConfig to CreateWorkflowOptions 
- Updated CreateWorkflowOptions to use dedicated types for WorkflowErrorConfig and WorkflowCompletedConfig

### CODE REVIEW

**Review by magnurud — CHANGES_REQUESTED** [review]
Lot's of great changes here! I have not looked at the tests yet, but it seems like 
2 empty files src/test/java/AssetsTest.java src/test/java/Service.java have been added without any content.

**Comment by augustak** [comment]
> Lot's of great changes here! I have not looked at the tests yet, but it seems like
> 2 empty files src/test/java/AssetsTest.java src/test/java/Service.java have been added without any content.

The files that are empty is being removed, not added

File: src/main/java/ai/lucidtech/las/sdk/Client.java [hunk 1]
```diff
@@ -178,11 +260,34 @@ public JSONObject updateAsset(
...
         UpdateAssetOptions options
     ) throws IOException, APIException, MissingAccessTokenException {
-        HttpUriRequest request = this.createAuthorizedRequest("PATCH", "/assets/" + assetId, options.toJson());
+        JSONObject body = new JSONObject();
+
+        if (options != null) {
+            body = options.addOptions(body);
+        }
+
+        HttpUriRequest request = this.createAuthorizedRequest("PATCH", "/assets/" + assetId, body);
```
[the remark this answers is not in the document]
Comment (augustak) [reply src/main/java/ai/lucidtech/las/sdk/Client.java:269]: I think all options objects should be allowed to be null. This is now consistent across the entire SDK.

File: src/main/java/ai/lucidtech/las/sdk/Client.java [hunk 2]
```diff
@@ -46,6 +45,90 @@ public Client(Credentials credentials) {
...
+     * @throws MissingAccessTokenException Raised if access token cannot be obtained
+     */
+    public JSONObject createAppClient(
+        CreateAppClientOptions options
+    ) throws IOException, APIException, MissingAccessTokenException {
+        JSONObject body = new JSONObject();
+
+        if (options != null) {
+            body = options.addOptions(body);
+        }
```
[the remark this answers is not in the document]
Comment (augustak) [reply src/main/java/ai/lucidtech/las/sdk/Client.java:65]: I think all options objects should be allowed to be null. This is now consistent across the entire SDK.

**Review by magnurud — APPROVED** [review]

**Pull request merged**